tests/integration/test_http_get_all.py

```diff
@@ -0,0 +1,88 @@
+# SPDX-License-Identifier: GPL-3.0-or-later
+
+import utils
+
+
+def test_http_get_all(test_env):
+    """
+    Check the API endpoint to get all requests.
+
+    Process:
+    * Pre-populate data in Cachito by submitting basic requests and waiting for them to complete.
+    * Verify the submitted requests are not incorrectly marked as "in_progress".
+    * Verify requests can be found using by using the state=complete filter.
+    Checks:
+    * Check that response code is 200
+    * Check that filtering parameters are working properly (page, per_page, and state)
+    """
+    client = utils.Client(test_env["api_url"], test_env["api_auth_type"])
+    requests_amount = test_env["http_get_all"]["requests_amount"]
+    submitted_requests = []
+    payload = {
+        "repo": test_env["package"]["repo"],
+        "ref": test_env["package"]["ref"],
+        "pkg_managers": test_env["package"]["pkg_managers"],
+    }
+    for _ in range(requests_amount):
+        initial_response = client.create_new_request(payload)
+        completed_response = client.wait_for_complete_request(initial_response)
+        response_data = completed_response.data
+        submitted_requests.append(response_data["id"])
```

Comment:
I realize my last comment was misleading since I showed the change in the for loop by mistake. Here is an example where you could take advantage of Cachito having multiple workers:
```suggestion
    submitted_requests = []
    initial_responses = [client.create_new_request(payload) for _ in range(requests_amount)]
    for initial_response in initial_responses
        client.wait_for_complete_request(initial_response)
        submitted_requests.append(initial_response["id"])
```

Reply:
Now it makes sense, thank you!